src/ucp/core/ucp_request.inl

```diff
@@ -433,9 +433,13 @@ ucp_request_send_state_advance(ucp_request_t *req,
                                const ucp_dt_state_t *new_dt_state,
                                unsigned proto, ucs_status_t status)
 {
-    if (status == UCS_ERR_NO_RESOURCE) {
-        /* Don't advance in order to continue
-         * on next try from last valid point. */
+    if (ucs_unlikely(UCS_STATUS_IS_ERR(status))) {
+        if (status != UCS_ERR_NO_RESOURCE) {
```

Comment:
lets handle `UCS_ERR_NO_RESOURCE` separately, it's more "likely" case vs other errors

Reply:
done